crates/ruff_linter/src/rules/flake8_builtins/rules/stdlib_module_shadowing.rs

```diff
@@ -23,6 +23,12 @@ use crate::settings::LinterSettings;
 /// Standard-library modules can be marked as exceptions to this rule via the
 /// [`lint.flake8-builtins.builtins-allowed-modules`] configuration option.
 ///
+/// By default, only the last component of the module name is considered, so `logging.py`,
+/// `utils/logging.py`, and `utils/logging/__init__.py` would all clash with the builtin `logging`
+/// module. With the [`lint.flake8-builtins.builtins-strict-checking`] option set to `false`, the
+/// module path is considered, so only a top-level `logging.py` or `logging/__init__.py` will
+/// trigger the rule and `utils/logging.py`, for example, would not.
+///
```

Comment:
We should document here or on the option the difference between preview and non preview. The way I remember it is that the default changes based on preview mode.